editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/features/create/ext/CreateFigureFeature.java

```diff
@@ -115,35 +125,80 @@ public CATEGORY_TYPE getCategoryType() {
+					// and check if the element can be added to a context
+					return NodeUtils.canBeAddedToCamelContextDirectly((AbstractCamelModelElement)obj);
+				}
 				return false;
 			}
-		} else if (containerBO instanceof AbstractCamelModelElement && ((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice")) {
+
+		// special handling for creating on choices (only one otherwise allowed)
+		} else if (	containerBO instanceof AbstractCamelModelElement && 
+					((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice")) {
 			// only one otherwise per choice
 			AbstractCamelModelElement choice = (AbstractCamelModelElement)containerBO;
+			
 			if (this.eip != null) {
-				if (this.eip.getName().equalsIgnoreCase("otherwise") && choice.getParameter("otherwise") != null) return false;
-				return this.eip.getName().equalsIgnoreCase("when") || this.eip.getName().equalsIgnoreCase("otherwise");
+				if (this.eip.getName().equalsIgnoreCase("otherwise") && 
+					choice.getParameter("otherwise") != null) {
+					// this choice already has an otherwise case - adding more isn't allowed
+					return false;
+				}
+				// allow when and otherwise children on choice
+				return this.eip.getName().equalsIgnoreCase("when") || 
+					   this.eip.getName().equalsIgnoreCase("otherwise");
+			
 			} else if (clazz != null) {
 				Object obj = newInstance(clazz);
-				if (obj instanceof AbstractCamelModelElement && ((AbstractCamelModelElement)obj).getNodeTypeId().equalsIgnoreCase("otherwise") && choice.getParameter("otherwise") != null) return false;
+				if (obj instanceof AbstractCamelModelElement && 
+					((AbstractCamelModelElement)obj).getNodeTypeId().equalsIgnoreCase("otherwise") && 
+					choice.getParameter("otherwise") != null) {
+					// this choice already has an otherwise case - adding more isn't allowed
+					return false;
+				}
 			}
 		}
 		
 		// we have to prevent some eips being dropped on the route even if camel catalog says its allowed
-		if (eip.getName().equalsIgnoreCase("when") || eip.getName().equalsIgnoreCase("otherwise") && containerBO != null && ((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice") == false) return false;
-		
-		return (containerBO != null && containerBO instanceof AbstractCamelModelElement && ((AbstractCamelModelElement)containerBO).getUnderlyingMetaModelObject().canHaveChildren() && ((AbstractCamelModelElement)containerBO).getUnderlyingMetaModelObject().getAllowedChildrenNodeTypes().contains(eip.getName()));
+		// probably issues in the catalog which will be fixed at later time, then we can
+		// remove the handling for these exceptions
+		if (eip.getName().equalsIgnoreCase("when") || eip.getName().equalsIgnoreCase("otherwise") && 
+			containerBO != null && 
+			!((AbstractCamelModelElement)containerBO).getNodeTypeId().equalsIgnoreCase("choice")) {
+			return false;
+		}
+		
+		if (containerBO != null && 
```

Comment:
containerBO instanceof AbstractCamelModelElement returns false is containerBO != null, so the first nullcheck can be removed